src/day8/re_frame/trace.cljs

```diff
@@ -215,43 +216,63 @@
                                           tag-str))]]
                                [:td.trace--meta.trace--details-icon
                                   {:on-click #(.log js/console tags)}]]))))))))
+
 (defn render-trace-panel []
   (let [filter-input               (r/atom "")
         filter-items               (r/atom (localstorage/get "filter-items" []))
         filter-type                (r/atom :contains)
         input-error                (r/atom false)
+        categories                 (r/atom #{:event :sub/run :sub/create})
         trace-detail-expansions    (r/atom {:show-all? false :overrides {}})]
     (add-watch filter-items
                :update-localstorage
                (fn [_ _ _ new-state]
                  (localstorage/save! "filter-items" new-state)))
     (fn []
-      (let [visible-traces       (if (= @filter-items [])
-                                   @traces
-                                   (filter (apply every-pred (map query->fn @filter-items)) @traces))
+      (let [toggle-category-fn   (fn [category-keys]
+                                   (swap! categories #(set (if (empty? (set/intersection % category-keys))
+                                                             (concat category-keys %)
+                                                             (remove category-keys %)))))
+            set-active           (fn [category]
+                                   (when (contains? @categories category) "active"))
```

Comment:
Minor: I think the name `set-active` could be confusing here, as it's not really setting anything. An alternative would be to provide a predicate `active?` and use `(when (active? :event) "active")` for the `:class` fields below. 

It's a bit more verbose but often it's easier to read a bunch of clojure core functions than one function that is specific to the problem at hand.

Reply:
True, I wasn't sure which alternative would have been preferred for this case. I will change it to
`{:class (when (contains? @categories :event) "active")` then for each category list item?